File: Lambda/src/main/java/me/denismurphy/tranquil/handlers/TranquilExecptionHandler.java
package me.denismurphy.tranquil.handlers;

import com.amazon.ask.dispatcher.exception.ExceptionHandler;
import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.model.Response;
import org.slf4j.Logger;
import com.amazon.ask.model.Request;
import com.amazon.ask.exception.AskSdkException;

import java.util.Optional;

import static org.slf4j.LoggerFactory.getLogger;

public class TranquilExecptionHandler implements ExceptionHandler {

    private static Logger LOGGER = getLogger(TranquilExecptionHandler.class);

    @Override
    public boolean canHandle(HandlerInput input, Throwable throwable) {
        return throwable instanceof AskSdkException;
    }

    @Override
    public Optional<Response> handle(HandlerInput input, Throwable throwable) {
        Request request = input.getRequestEnvelope().getRequest();
        LOGGER.error("Request: " + request.toString());
        LOGGER.error("Stacktrace", throwable);
        return input.getResponseBuilder()
                .withShouldEndSession(true)
                .build();
    }
}

File: lambda/src/main/kotlin/me/denismurphy/tranquil/handlers/TranquilExecptionHandler.kt
package me.denismurphy.tranquil.handlers

import com.amazon.ask.dispatcher.exception.ExceptionHandler
import com.amazon.ask.dispatcher.request.handler.HandlerInput
import com.amazon.ask.exception.AskSdkException
import com.amazon.ask.model.Response
import org.slf4j.LoggerFactory.getLogger
import java.util.*

class TranquilExecptionHandler : ExceptionHandler {

    override fun canHandle(input: HandlerInput, throwable: Throwable): Boolean {
        return throwable is AskSdkException
    }

    override fun handle(input: HandlerInput, throwable: Throwable): Optional<Response> {
        val request = input.requestEnvelope.request
        LOGGER.error("Request: $request")
        LOGGER.error("Stacktrace", throwable)
        return input.responseBuilder
                .withShouldEndSession(true)
                .build()
    }

    companion object {
        private val LOGGER = getLogger(TranquilExecptionHandler::class.java)
    }
}